### DISCUSSION TITLE
Email loop after upgrade from 3.4.0 to 4.6.5-BETA

### DISCUSSION THREAD

**Discussion by Metabaron1** [simple-login/app#1632] [Self-hosting question] [answered]
Hail

I'm still testing upgrade  to 4.6.5-BETA, and now I'm facing a problem: emails sent to aliases are not forwarded to mailbox.

I updated DB:
`simplelogin=# update email_log set alias_id=(select alias_id from contact where contact.id = email_log.contact_id);`

I replaced my old postfix container with your simplelogin-postfix-docker:4.1.0 with RELAY_HOST and SIMPLELOGIN_COMPATIBILITY_MODE=v4 (/etc/postfix/pgsql-relay-domains.cf and /etc/postfix/pgsql-transport-maps.cf are generated correctly).

I  checked https://github.com/simple-login/app/blob/master/docs/troubleshooting.md : 

Postfix can connect to sl-email
```
docker exec -ti sl-postfix /bin/sh
# nc sl-email 20381
220 9b14c08e75e1 Python SMTP 1.4.2
```

 sl-email can connect to Postfix
```
docker exec -ti sl-email /bin/sh
# nc sl-postfix 25
220 mydomain.fr ESMTP Postfix
```

 sl-app can connect to Postfix
```
docker exec -ti sl-app /bin/sh
# nc sl-postfix 25
220 mydomain.fr ESMTP Postfix
```

Postfix domain is OK:
```
docker exec -ti sl-postfix /bin/sh
/ # postmap -q mydomain.fr pgsql:/etc/postfix/pgsql-relay-domains.cf
mydomain.fr
/ # postmap -q not-exist.com pgsql:/etc/postfix/pgsql-relay-domains.cf
/ # postmap -q mydomain.fr pgsql:/etc/postfix/pgsql-transport-maps.cf
smtp:sl-email:20381
/ # postmap -q not-exist.com pgsql:/etc/postfix/pgsql-transport-maps.cf
/ # 

```

I can send email from Postfix (via relay host)
```
docker exec -ti sl-postfix /bin/sh
/src # swaks --to mymailbox@free.fr -s smtp.free.fr:25 -f root@mydomain.fr
=== Trying smtp.free.fr:25...
=== Connected to smtp.free.fr.
<-  220 smtp3-g21.free.fr ESMTP Postfix
 ...
 -> This is a test mailing
 -> 
 -> 
 -> .
<-  250 2.0.0 Ok: queued as 212CA13F86F
 -> QUIT
<-  221 2.0.0 Bye
=== Connection closed with remote host.
```

I  checked DB data:
`simplelogin=# select alias.email,alias.mailbox_id , mailbox.email from alias,mailbox where alias.email like 'root@%' and alias.mailbox_id=mailbox.id;
      email      | mailbox_id |       email        
-----------------+------------+--------------------
 root@mydomain.fr |          1 | mymailbox@free.fr
(1 row)`

and finally when testing sending email from a test@email.fr  (different from mymailbox@free.fr of course) to root@mydomain.fr
I can see Postfix log receiving the email.
But sl-email shows a loop error email sent ,  it should not as test@email.fr is different from mymailbox@free.fr.
And the error email is not event sent:
```

==> sl-email.log <==
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,146 - SL - DEBUG - 1 - "/code/app/log.py:24" - set_message_id() - 8b53264f-2825-41ab-852b-b889435223a1 - set message_id 34c39408-9e9d-4f36-b200-c841f6dcc9f4
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,146 - SL - DEBUG - 1 - "/code/email_handler.py:2379" - _handle() - 34c39408-9e9d-4f36-b200-c841f6dcc9f4 - ====>=====>====>====>====>====>====>====>
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,147 - SL - INFO - 1 - "/code/email_handler.py:2380" - _handle() - 34c39408-9e9d-4f36-b200-c841f6dcc9f4 - New message, mail from test@email.fr, rctp tos ['root@mydomain.fr'] 
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,151 - SL - INFO - 1 - "/code/email_handler.py:1995" - handle() - 34c39408-9e9d-4f36-b200-c841f6dcc9f4 - Set CONTENT_TRANSFER_ENCODING
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,151 - SL - DEBUG - 1 - "/code/app/log.py:24" - set_message_id() - 34c39408-9e9d-4f36-b200-c841f6dcc9f4 - set message_id E8DC526ED
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,157 - SL - DEBUG - 1 - "/code/email_handler.py:2018" - handle() - E8DC526ED - ==>> Handle mail_from:test@email.fr, rcpt_tos:['root@mydomain.fr'], header_from:test <test@email.fr>, header_to:root <root@mydomain.fr>,...
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,172 - SL - DEBUG - 1 - "/code/email_handler.py:2239" - handle() - E8DC526ED - Forward phase test@email.fr(test <test@email.fr>) -> root@mydomain.fr
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,200 - SL - INFO - 1 - "/code/email_handler.py:620" - handle_forward() - E8DC526ED - cycle email sent from test@email.fr to <Alias 2 root@mydomain.fr>
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,335 - SL - DEBUG - 1 - "/code/email_handler.py:549" - handle_email_sent_to_ourself() - E8DC526ED - Create refused email <Refused Email 579 None 2023-03-18T22:38:16.204987+00:00>
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,539 - SL - WARNING - 1 - "/code/app/email_utils.py:415" - send_email_at_most_times() - E8DC526ED - 1 emails were sent to test@email.fr alert type cycle
Mar 11 23:38:16 172.22.0.1 sl-email[28194]: 2023-03-11 22:38:16,539 - SL - INFO - 1 - "/code/email_handler.py:2404" - _handle() - E8DC526ED - Finish mail_from test@email.fr, rcpt_tos ['root@mydomain.fr'], takes 0.3933901786804199 seconds with return code '250 SL E209 Email Loop'<<===

==> sl-postfix.log <==
Mar 11 23:38:16 172.22.0.1 sl-postfix[28194]: Mar 11 23:38:16 mydomain postfix/smtp[274]: E8DC526ED: to=<root@mydomain.fr>, relay=sl-email[172.18.0.4]:20381, delay=0.78, delays=0.35/0.03/0/0.4, dsn=2.0.0, status=sent (250 SL E209 Email Loop)
```

I have tested with another alias than root@mydomain.fr and seems that all aliases are affected.
How can I manually check the test that triggers a loop email?
What else should I check?

**Comment by nguyenkims** [comment]
Hi the loop is triggered when an email sent from one of user's mailboxes to an alias. I wonder if that can be the case here?

**Comment by Metabaron1** [comment, marked as answer]
No the email used to test send mail to aliases  is not a users's  mailbox. I use the same email to test with 3.4.0 and there is no loop.
Coud you tell me witch SQL is used for the loop test please, so I can check the result

**Reply by nguyenkims** [reply]
The check happens here https://github.com/simple-login/app/blob/44e0dd8635db041f4df33d4eaf42cd021c31e87d/email_handler.py#L637

It also checks for "authorized addresses", any chance you have the sender listed as authorized address?

**Comment by Metabaron1** [comment]
Yes it was autorized adress! :)

I had seen the code but I had no idea what was authorized_addresses.
By searching I found the parameter, and indeed, it it works after deleting.

No idea why I configured that a few years ago... Definitely for testing. and since it doesn't trigger any error in 3.4.0, I ended up forgetting it ;) But I undertant the update.

I think this BETA is fully working now :)
Thanks again for this tool and for your help and time @nguyenkims

**Reply by nguyenkims** [reply]
Cool, glad that everything works now!
